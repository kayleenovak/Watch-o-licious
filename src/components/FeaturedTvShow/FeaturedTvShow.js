import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import './FeaturedTvShow.css'

export const FeaturedTvShow = (props) => {
    const {title, runtime, premiered, rating, summary, image, id} = props
    return (
      <section className='featured-tv-show'>
        <article className='featured-info'>
          <h3 className='featured-title'>{title}</h3>
          <p className='featured-summary'>{summary}</p>
          <p className='featured-runtime'>Runtime: {runtime} minutes</p>
          <p className='featured-date'>Premiere date: {premiered}</p>
          <p className='featured-rating'>{rating}</p>
          <Link to={`/tvshow/${id}`} key={id}>
            <button className='view-series-btn'>View Series</button>
          </Link>
        </article>
        <img alt='series-poster' className='featured-img' src={image} />
      </section>
    )
}

FeaturedTvShow.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  runtime: PropTypes.number,
  premiered: PropTypes.string,
  schedule: PropTypes.object,
  network: PropTypes.string,
  summary: PropTypes.string,
  image: PropTypes.string
}

