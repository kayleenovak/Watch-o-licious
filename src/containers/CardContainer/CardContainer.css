.card-container {
  max-width: 100vw;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  height: 100%;
}