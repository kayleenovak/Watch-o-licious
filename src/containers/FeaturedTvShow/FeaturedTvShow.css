.featured-tv-show {
  display: flex;
  justify-content: center;
  margin: 10px 0 50px 0;
}

.featured-info {
  width: 600px;
}

.featured-title {
  font-size: 35px;
  margin: 0 0 15px 0;
  font-family: 'Montserrat', sans-serif;
  color: white;
  margin-left: 5px;
}

.featured-summary {
  font-family: 'Raleway', sans-serif;
  border-radius: 3px;
  cursor: default;
  overflow: auto;
  max-height: 150px;
  transition: border 500ms;
  padding: 5px;
  border: 2px solid transparent;
  color: white;
}

.featured-runtime,
.featured-date,
.featured-rating {
  margin-left: 5px;
  color: white;
  font-family: 'Raleway', sans-serif;

}

.featured-summary:hover {
  border: 2px solid #AEA7B9;
  box-shadow: inset 0 -45px 130px -55px #AEA7B9;
}

.featured-img {
  height: 354px;
  width: 252px;
  margin-left: 25px;
  box-shadow: -2px 4px 25px black;
}
