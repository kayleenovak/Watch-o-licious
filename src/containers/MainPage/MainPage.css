.main-page {
  display: grid;
  grid-template-rows: 300px 1fr;
}
