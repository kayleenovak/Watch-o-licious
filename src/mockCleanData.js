export const mockCleanCakeShows = [
  {
    id: 921,
    title: 'Cake Boss',
    runtime: 30,
    premiered: '2009-05-25',
    schedule: {
      'time': '21:00',
      'days': [
        'Monday'
      ]
    },
    rating: null,
    network: 'TLC',
    summary: 'Cake Boss is a popular docu-series following beloved baker Buddy Valastro, his larger-than-life family and burgeoning businesses.',
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/6/16129.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/6/16129.jpg"
  },
  {
    id: 27786,
    title: 'Extreme Cake Makers',
    runtime: 30,
    premiered: '2017-05-01',
    schedule: {
        time: "17:30",
        days: [
          "Monday",
          "Tuesday",
          "Wednesday",
          "Thursday",
          "Friday"
      ]
    },
    rating: null,
    network: 'Channel 4',
    summary: "Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg"
  }
]

export const mockCleanBakeShows = [
    {
    id: 2950,
    title: 'The Great British Bake Off',
    runtime: 75,
    premiered: '2010-08-17',
    schedule: {
      time: "20:00",
      days: [
        "Tuesday"
      ]
    },
    rating: 8.7,
    network: 'Channel 4',
    summary: "Over 10 weeks, 12 of the best amateur bakers in Britain will whisk, knead, ice, beat and bake their way through classic British cakes, perfect patisserie, Italian delights, sticky caramel constructions and elaborate layered puddings. All 12 will be hoping to impress with their skill, creativity, knowledge and passion to clinch the Bake Off Crown. Each of the 30 new challenges have been carefully designed by judges Paul Hollywood and Prue Leith to reveal just who is a star baker. But it's not just a new experience for the bakers. Also joining the tent for the first time are Sandi Toksvig and Noel Fielding, who will be with the bakers every step of the way, ready with a pertinent pep talk, a helpful hand or just a sympathetic shoulder to cry on.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/124/311372.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/124/311372.jpg"
  },
    {
    id: 39695,
    title: 'The Great Kiwi Bake Off',
    runtime: 60,
    premiered: '2018-10-16',
    schedule: {
      time: "19:30",
      days: [
        "Tuesday"
      ]
    },
    rating: null,
    network: 'TVNZ 2',
    summary: "From biscuits to bread, pies to pavlovas, 12 amateur Kiwi bakers have their skills put to the test. Will they rise to the challenge?",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/174/435768.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/174/435768.jpg"
  }
]

export const mockCleanBakingShows = [
  {
    id: 1549,
    title: 'Kids Baking Championship',
    runtime: 60,
    premiered: '2015-02-02',
    schedule: {
      time: "21:00",
      days: [
          "Monday"
      ]
    },
    rating: null,
    network: 'Food Network',
    summary: "Hosts and judges Duff Goldman and Valerie Bertinelli lead eight talented kid bakers through tasty challenges designed to find the most impressive and creative young baker on Kids Baking Championship. In each challenge, the kid contestants (ranging in age from 10 to 13) will be tested on their baking skills and originality, as they whip up delectable desserts including creative cakes, mouthwatering cookies and pastries making it seem as easy as pie. At stake, a sweet prize package that includes a complete set of Food Network bakeware products, a profile in Food Network Magazine, an original cake featured at Charm City Cakes and a $10,000 grand prize. Only one will take the cake to become the Kids Baking Champion!",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/88/220914.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/88/220914.jpg"
  },
  {
    id: 6407,
    title: 'Holiday Baking Championship',
    runtime: 60,
    premiered: '2014-11-09',
    schedule: {
      time: "21:00",
      days: [
        "Monday"
      ]
    },
    rating: null,
    network: 'Food Network',
    summary: "On Holiday Baking Championship the search to find the greatest holiday baker begins as eight amateur bakers enter the kitchen, where they show off their family traditions and superb baking skills. To survive the challenges from week to week, they must prove their abilities in front of our tough-love judges: Duff Goldman, Nancy Fuller and Lorraine Pascale. Hosted by Bobby Deen, this baking championship is the sweetest way to celebrate this holiday season.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/24/61290.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/24/61290.jpg"
  }
]