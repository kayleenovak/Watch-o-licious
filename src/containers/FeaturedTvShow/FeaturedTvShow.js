import React, { Component } from 'react'
import { connect } from 'react-redux'
import './FeaturedTvShow.css'

export default class FeaturedTvShow extends Component {

  render() {
    const {title, runtime, premiered, rating, summary, image} = this.props
    return (
      <section className='featured-tv-show'>
        <article className='featured-info'>
          <h3 className='featured-title'>{title}</h3>
          <p className='featured-summary'>{summary}</p>
          <p className='featured-runtime'>Runtime: {runtime} minutes</p>
          <p className='featured-date'>Premiere date: {premiered}</p>
          <p className='featured-rating'>{rating}</p>
        </article>
        <img className='featured-img' src={image} />
      </section>
    )
  }
}

