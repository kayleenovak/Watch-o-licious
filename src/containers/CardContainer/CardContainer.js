import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import TvShowCard from '../TvShowCard/TvShowCard'
import './CardContainer.css'
import FavoriteCard from '../FavoriteCard/FavoriteCard.js'

export class CardContainer extends Component {

  shuffleTvShows = (tvShows) => {
    for (let i = tvShows.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [tvShows[i], tvShows[j]] = [tvShows[j], tvShows[i]];
    }
    return tvShows
  }

  splitLocation = (location) => {
    const splitLocation = location.split('/')
    return splitLocation[1]
  }
  
  render() {
    const { location } = this.props
    if(location === '/favorites' || location === '/watched' || location === '/watchlist') {
      const locationString = this.splitLocation(location)
      const trackedEpisodes = this.props.trackedEpisodes.filter(episode => {
        return episode.tracked[locationString] === true
      }).map(episode => {
        return <FavoriteCard />
      })
      return (
        <section className='tracked-episodes'>
          {trackedEpisodes}
        </section>
      )
    } else {
      const shuffledTvShows = this.shuffleTvShows(this.props.tvShows)
      const tvCards = shuffledTvShows.map(tvShow => {
        return <TvShowCard {...tvShow} />
      })
      return (
        <section className='card-container'>
          {tvCards}
        </section>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  tvShows: state.tvShows,
  trackedEpisodes: state.tracked
})

export default withRouter(connect(mapStateToProps)(CardContainer))