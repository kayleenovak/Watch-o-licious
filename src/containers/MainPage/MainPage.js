import React, { Component } from 'react'
import { connect } from 'react-redux'
import { FeaturedTvShow } from '../../components/FeaturedTvShow/FeaturedTvShow.js'
import CardContainer from '../CardContainer/CardContainer.js'
import { Loading } from '../../components/Loading/Loading.js'


export class MainPage extends Component {

  randomFeaturedTvShow = () => {
    const max = this.props.tvShows.length - 1
    const min = 0
    const randomNumber = Math.floor(Math.random() * (max - min + 1)) + min
    return <FeaturedTvShow {...this.props.tvShows[randomNumber]} />
  }

  render() {
    if(this.props.isLoading === true) {
      return <Loading />
    } else if (this.props.tvShows){
      const randomTvShow = this.randomFeaturedTvShow()
      return (
        <div className='main-page'>
          {
            randomTvShow
          }
          <CardContainer />
        </div>
      )
    } else if (this.props.hasErrored === true) {
      return <div>Error</div>
    } else {
      return <div>Error</div>
    }
  }
}

export const mapStateToProps = (state) => ({
  tvShows: state.tvShows,
  isLoading: state.isLoading,
  hasErrored: state.hasErrored
})

export default connect(mapStateToProps)(MainPage)
