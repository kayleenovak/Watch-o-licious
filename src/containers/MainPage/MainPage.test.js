import React from 'react'
import { shallow } from 'enzyme'
import { MainPage, mapStateToProps } from './MainPage.js'
import FeaturedTvShow from '../FeaturedTvShow/FeaturedTvShow.js'
import CardContainer from '../CardContainer/CardContainer.js'
import { mockCleanCakeShows } from '../../mockCleanData.js'

describe('MainPage', () => {
  it('should render a FeaturedTvShow and CardContainer component if there are tvShows', () => {
    const wrapper = shallow(<MainPage tvShows={ mockCleanCakeShows } />)

    expect(wrapper.find(FeaturedTvShow).length).toEqual(1)
    expect(wrapper.find(CardContainer).length).toEqual(1)
  })

  it('should not render FeaturedTvShow or CardContainer if there are no tvShows', () => {
    const wrapper = shallow(<MainPage />)

    expect(wrapper.find(FeaturedTvShow).length).toEqual(0)
    expect(wrapper.find(CardContainer).length).toEqual(0)
  })

  describe('mapStateToProps', () => {
    it('should return an object with tvShows as a key', () => {
      const mockState = {
        tvShows: mockCleanCakeShows,
        isLoading: false,
        hasErrored: false
      }
      const expected = {
        tvShows: mockCleanCakeShows
      }
      
      const result = mapStateToProps(mockState)

      expect(result).toEqual(expected)
    })
  })
})