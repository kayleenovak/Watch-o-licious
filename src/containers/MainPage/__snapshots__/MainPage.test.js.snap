// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    hasErrored={false}
    isLoading={false}
    tvShows={
      Array [
        Object {
          "id": 921,
          "image": "http://static.tvmaze.com/uploads/images/medium_portrait/6/16129.jpg",
          "largeImage": "http://static.tvmaze.com/uploads/images/original_untouched/6/16129.jpg",
          "network": "TLC",
          "premiered": "2009-05-25",
          "rating": null,
          "runtime": 30,
          "schedule": Object {
            "days": Array [
              "Monday",
            ],
            "time": "21:00",
          },
          "summary": "Cake Boss is a popular docu-series following beloved baker Buddy Valastro, his larger-than-life family and burgeoning businesses.",
          "title": "Cake Boss",
        },
        Object {
          "id": 27786,
          "image": "http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg",
          "largeImage": "http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg",
          "network": "Channel 4",
          "premiered": "2017-05-01",
          "rating": null,
          "runtime": 30,
          "schedule": Object {
            "days": Array [
              "Monday",
              "Tuesday",
              "Wednesday",
              "Thursday",
              "Friday",
            ],
            "time": "17:30",
          },
          "summary": "Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth.",
          "title": "Extreme Cake Makers",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FeaturedTvShow
          id={27786}
          image="http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg"
          largeImage="http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg"
          network="Channel 4"
          premiered="2017-05-01"
          rating={null}
          runtime={30}
          schedule={
            Object {
              "days": Array [
                "Monday",
                "Tuesday",
                "Wednesday",
                "Thursday",
                "Friday",
              ],
              "time": "17:30",
            }
          }
          summary="Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth."
          title="Extreme Cake Makers"
        />,
        <withRouter(Connect(CardContainer)) />,
      ],
      "className": "main-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": 27786,
          "image": "http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg",
          "largeImage": "http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg",
          "network": "Channel 4",
          "premiered": "2017-05-01",
          "rating": null,
          "runtime": 30,
          "schedule": Object {
            "days": Array [
              "Monday",
              "Tuesday",
              "Wednesday",
              "Thursday",
              "Friday",
            ],
            "time": "17:30",
          },
          "summary": "Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth.",
          "title": "Extreme Cake Makers",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FeaturedTvShow
            id={27786}
            image="http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg"
            largeImage="http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg"
            network="Channel 4"
            premiered="2017-05-01"
            rating={null}
            runtime={30}
            schedule={
              Object {
                "days": Array [
                  "Monday",
                  "Tuesday",
                  "Wednesday",
                  "Thursday",
                  "Friday",
                ],
                "time": "17:30",
              }
            }
            summary="Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth."
            title="Extreme Cake Makers"
          />,
          <withRouter(Connect(CardContainer)) />,
        ],
        "className": "main-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": 27786,
            "image": "http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg",
            "largeImage": "http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg",
            "network": "Channel 4",
            "premiered": "2017-05-01",
            "rating": null,
            "runtime": 30,
            "schedule": Object {
              "days": Array [
                "Monday",
                "Tuesday",
                "Wednesday",
                "Thursday",
                "Friday",
              ],
              "time": "17:30",
            },
            "summary": "Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth.",
            "title": "Extreme Cake Makers",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
